2019-May-Season/SoftUni-Information-Services/src/SIS.HTTP/Headers/HttpHeaderCollection.cs

```diff
@@ -1,39 +1,61 @@
-﻿using System.Collections.Generic;
-using System.Linq;
-using SIS.HTTP.Common;
-using SIS.HTTP.Headers.Contracts;
-
-namespace SIS.HTTP.Headers
+﻿namespace SIS.HTTP.Headers
 {
-    public class HttpHeaderCollection : IHttpHeaderCollection
+    using System.Linq;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    using SIS.HTTP.Common;
+    using SIS.HTTP.Headers.Contracts;
+    public class HttpHeaderCollection : IHttpHeaderCollection, IEnumerable<IHttpHeader>
     {
-        private Dictionary<string, HttpHeader> httpHeaders;
+        private readonly IList<IHttpHeader> httpHeaders;
```

Comment:
Again using a IList<T> defeats the purpose of using the dictionary's key hashtable for searching...